app/main.py

```diff
@@ -1,6 +1,43 @@
+import os
+
+import requests
+from requests import RequestException
+
+
+URL = "http://api.weatherapi.com/v1/current.json"
+API_KEY = os.environ["API_KEY"]
+FILTERING = "Paris"
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    try:
+        response = requests.get(URL + f"?key={API_KEY}&q={FILTERING}")
+    except ConnectionError:
+        raise ConnectionError(
+            "There was a problem connecting to the resource."
+        )
+    except RequestException:
+        raise RequestException("An error occurred while making the request.")
+
+    if response.status_code == 200:
+        try:
+            city = response.json().get("location").get("name")
+            country = response.json().get("location").get("country")
+            localtime = response.json().get("location").get("localtime")
+            celsius = response.json().get("current").get("temp_c")
+            condition = response.json().get(
+                "current"
+            ).get("condition").get("text")
+        except ValueError:
+            raise ValueError("Failed to decode the JSON response.")
+        except KeyError:
+            raise KeyError("Missing expected data in the response.")
+    else:
+        print("Failed to retrieve required data.")
```

Comment:
Consider adding more detailed error handling for unsuccessful API requests. For example, you could print the status code or the response content to help diagnose issues.